### PR TITLE
Issue #4937: Allow to upload webp to hero blocks if supported

### LINKED ISSUE #4937: Allow webp extension for hero block image upload (if supported)
**Description of the bug**

This popped up [in the forum](https://forum.backdropcms.org/forum/webp). Webp support is in core since 1.17.5.
However, the list of allowed extensions in hero blocks is currently hardcoded and doesn't consider the latest change.

https://github.com/backdrop/backdrop/blob/ac819913fdb7aadcec9c7579fd33920327526792/core/modules/layout/includes/block.hero.inc#L56

**Proposed solution**

Either add webp to that hardcoded list (only if supported, of course), or provide a core function for "allowed image extensions", which modules can re-use.

### PR DESCRIPTION
**Pull request by indigoxela**
Fixes https://github.com/backdrop/backdrop-issues/issues/4937

### CODE REVIEW

File: core/modules/layout/includes/block.hero.inc [hunk 1]
```diff
@@ -52,8 +52,15 @@ class BlockHero extends BlockText {
   function form(&$form, &$form_state) {
     parent::form($form, $form_state);
 
+    $allowed_extensions = 'jpg jpeg png gif';
+    if (config_get('system.core', 'image_toolkit') == 'gd' && defined('IMAGETYPE_WEBP')) {
+      $gd_info = gd_info();
+      if (isset($gd_info['WebP Support']) && $gd_info['WebP Support'] == TRUE) {
```
Comment (docwilmot) [inline core/modules/layout/includes/block.hero.inc]: Same as `if (!empty($gd_info['WebP Support']))`.

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

**Review by yorkshire-pudding — APPROVED** [review]
Looks good to me

**Comment by indigoxela** [comment]
Closing as outdated and in favor of a more generic approach.

**Pull request closed without merging**